packages/components/src/internal/components/domainproperties/DomainForm.tsx

```diff
@@ -334,48 +356,65 @@ export class DomainFormImpl extends React.PureComponent<IDomainFormInput, IDomai
         // note: we skip any rowIndexChange which has a newIndex as those are just reorder changes
-        if (rowIndexChange?.newIndex === undefined && ontologyLookupIndices.length > 0) {
-            ontologyLookupIndices.forEach(index => {
-                // skip any ontology lookup fields if they were removed
-                const ontFieldRemoved = rowIndexChange?.originalIndex === index;
-
-                if (!ontFieldRemoved) {
-                    updatedDomain = updateOntologyFieldProperties(
-                        // check for a field removal prior to the ontology lookup field
-                        rowIndexChange?.originalIndex < index ? index - 1 : index,
-                        domainIndex,
-                        updatedDomain,
-                        domain,
-                        rowIndexChange?.originalIndex
-                    );
+        if (rowIndexChanges) {
+            rowIndexChanges.forEach((rowIndexChange) => {
+                if (rowIndexChange?.newIndex === undefined && ontologyLookupIndices.length > 0) {
+                    ontologyLookupIndices.forEach(index => {
+                        // skip any ontology lookup fields if they were removed
+                        const ontFieldRemoved = rowIndexChange?.originalIndex === index;
+
+                        if (!ontFieldRemoved) {
+                            updatedDomain = updateOntologyFieldProperties(
+                                // check for a field removal prior to the ontology lookup field
+                                rowIndexChange?.originalIndex < index ? index - 1 : index,
+                                domainIndex,
+                                updatedDomain,
+                                domain,
+                                rowIndexChange?.originalIndex
+                            );
+                        }
+                    });
                 }
             });
         }
 
-        this.setState(() => ({ fieldDetails: updatedDomain.getFieldDetails() }));
+        const visibleFieldsCount = getVisibleFieldCount(updatedDomain);
+        this.setState(() => ({ fieldDetails: updatedDomain.getFieldDetails(), visibleFieldsCount }));
 
-        this.props.onChange?.(updatedDomain, dirty !== undefined ? dirty : true, rowIndexChange);
+        this.props.onChange?.(updatedDomain, dirty !== undefined ? dirty : true, rowIndexChanges);
     }
 
-    onDeleteConfirm(index: number) {
+    clearFilePreviewData = (currentState: Partial<IDomainFormState>) => {
         let fieldCount = this.props.domain.fields.size;
+
+        this.setState({
+            filePreviewData: fieldCount === 0 ? undefined : currentState.filePreviewData,
+            file: fieldCount === 0 ? undefined : currentState.file,
+            filePreviewMsg: undefined,
+        });
+    }
+
+    onDeleteConfirm(index: number) {
+        let visibleSelection;
+        let visibleFieldsCount;
+        let selectAll;
         if (index !== undefined) {
```

Comment:
As mentioned in other comment. I don't think the index here can actually be `undefined` so if it is removed then these three variables will always be defined in this function before calling `this.setState()`.

Reply:
Done!